apps/web/test/utils/bookingScenario.ts

```diff
@@ -0,0 +1,742 @@
+    const users =
+      eventType.users?.map((userWithJustId) => {
+        return usersStore.find((user) => user.id === userWithJustId.id);
+      }) || [];
+    return {
+      ...baseEventType,
+      ...eventType,
+      workflows: [],
+      users,
+    };
+  });
+
+  logger.silly("TestData: Creating EventType", eventTypes);
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  const eventTypeMock = ({ where }) => {
+    return new Promise((resolve) => {
+      const eventType = eventTypesWithUsers.find((e) => e.id === where.id) as unknown as PrismaEventType & {
+        users: PrismaUser[];
+      };
+      resolve(eventType);
+    });
+  };
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  prismaMock.eventType.findUnique.mockImplementation(eventTypeMock);
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  prismaMock.eventType.findUniqueOrThrow.mockImplementation(eventTypeMock);
+}
+
+async function addBookings(bookings: InputBooking[], eventTypes: InputEventType[]) {
+  logger.silly("TestData: Creating Bookings", bookings);
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  prismaMock.booking.findMany.mockImplementation((findManyArg) => {
+    const where = findManyArg?.where || {};
+    return new Promise((resolve) => {
+      resolve(
+        bookings
+          // We can improve this filter to support the entire where clause but that isn't necessary yet. So, handle what we know we pass to `findMany` and is needed
+          .filter((booking) => {
+            /**
+             * A user is considered busy within a given time period if there
+             * is a booking they own OR host. This function mocks some of the logic
+             * for each condition. For details see the following ticket:
+             * https://github.com/calcom/cal.com/issues/6374
+             */
+
+            // ~~ FIRST CONDITION ensures that this booking is owned by this user
+            //    and that the status is what we want
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            const statusIn = where.OR[0].status?.in || [];
+            const firstConditionMatches =
+              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+              // @ts-ignore
+              statusIn.includes(booking.status) && booking.userId === where.OR[0].userId;
+
+            // We return this booking if either condition is met
+            return firstConditionMatches;
+          })
+          .map((booking) => ({
+            uid: uuidv4(),
+            title: "Test Booking Title",
+            ...booking,
+            eventType: eventTypes.find((eventType) => eventType.id === booking.eventTypeId),
+          })) as unknown as PrismaBooking[]
+      );
+    });
+  });
+}
+
+async function addWebhooks(webhooks: InputWebhook[]) {
```

Comment:
Do we need `async` here?

```suggestion
function addWebhooks(webhooks: InputWebhook[]) {
```

Reply:
Nope, will fix it in the stacked PR.